common/src/services/cipher.service.ts

```diff
@@ -303,22 +309,95 @@ export class CipherService implements CipherServiceAbstraction {
             return this.decryptedCipherCache;
         }
 
-        const decCiphers: CipherView[] = [];
         const hasKey = await this.cryptoService.hasKey();
         if (!hasKey) {
             throw new Error('No key.');
         }
 
-        const promises: any[] = [];
         const ciphers = await this.getAll();
+
+        this.consoleLogService.time('getAllDecrypted');
+        let decryptedCiphers;
+        if (this.platformUtilsService.supportsWebWorkers() && ciphers.length > this.workerThreshold) {
+            this.consoleLogService.info('getAllDecrypted started using worker');
+            const userId = await this.userService.getUserId();
+            const cipherData = ciphers.map(c => c.toCipherData(userId));
```

Comment:
Why do we have to turn it back to cipherData? because it needs to be serializable or something?